pkg/nfd-worker/nfd-worker.go

```diff
@@ -271,6 +273,36 @@ func (w *nfdWorker) Run() error {
 	labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
 	defer labelTrigger.Stop()
 
+	// Get pod owner reference
+	podName := os.Getenv("POD_NAME")
+	client, err := w.getKubeClient()
+	if err != nil {
+		return fmt.Errorf("failed to get kube client: %w", err)
+	}
+
+	selfPod, err := client.CoreV1().Pods(w.kubernetesNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get pod %q: %w", podName, err)
+	}
+
+	// Create owner ref
+	ownerReference := selfPod.OwnerReferences
+
+	// Add pod owner reference if it exists
+	podUID := os.Getenv("POD_UID")
+	if podName != "" && podUID != "" {
+		isTrue := true
+		ownerReference = append(ownerReference, metav1.OwnerReference{
+			APIVersion: "v1",
+			Kind:       "Pod",
+			Name:       podName,
+			UID:        types.UID(podUID),
+			Controller: &isTrue,
```

Comment:
Now that we're at it we could just drop this, I think. It doesn't affect the garbage collection and nobody's actually looking at this

```suggestion
			Controller: &isTrue,
```

alternatively, use `ptr.To` from `k8s.io/utils`

```suggestion
			Controller: ptr.To[bool](true),
```